routes/api/follows.js

```diff
@@ -1,152 +1,206 @@
-const express = require('express');
+const express = require("express");
 const router = express.Router();
-const passport = require('passport');
-const User = require('../../models/User');
+const passport = require("passport");
+const User = require("../../models/User");
 
 // @route   put api/follow/:userId
 // @desc    Follow User
 // @input   Get Userid from req params of person whom user wants to follow.
 // @access  Private
-
-
-router.put('/:userId/follow', passport.authenticate('jwt', { session: false }), (req, res) => {
-  const userId = req.params.userId;
-  // check if your id doesn't match the id of the user you want to follow
-  if (req.user.id === userId) {
-    return res.status(400).json({ alreadyfollow: "You cannot follow yourself" })
-  }
-  User.findById(userId)
-    .then(user => {
-      if (user) {//user exists
-        // check if the requested user is already in follower list of other user then 
-
-        if (user.followers.some(follower => follower.user.toString() === req.user.id)) {
-          return res.status(400).json({ alreadyfollow: "You already followed the user" });
-        }
-        user.followers.unshift({ user: req.user.id });//adding in follower[]
-        user.save()
-          .then(result => {
-            User.findOne({ email: req.user.email }) 
-              .then(user => {
-                if (user.following.filter(following =>
-                  following.user.toString() === userId).length > 0) {
-                  return res.status(400).json({ alreadyfollow: "You already following the user" })
+router.put(
+  "/:user_id/follow",
+  passport.authenticate("jwt", { session: false }),
+  (req, res) => {
+    // check if your id doesn't match the id of the user you want to follow
+    if (req.user.id === req.params.user_id) {
```

Comment:
I added  const userId = req.params.userId; in PR 19. And every where I used that userId instead of directly using  req.params.userId
I did it for Both follow and unfollow. That is missing here. If there are merge conflicts, then retain master code for follow and unfollow

Reply:
okay, when I merge I will keep `const userId = req.params.userId;`

Reply:
ok